core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyCommand.java

```diff
@@ -18,6 +18,8 @@
 import org.infinispan.functional.impl.EntryViews;
 import org.infinispan.functional.impl.Params;
 
+// TODO: the command does not carry previous values to backup, so it can cause
+// the values on primary and backup owners to diverge in case of topology change
```

Comment:
@rvansa That's right. It was implemented partially because the solution is not optimal.